### PR TITLE
Initial php 8 compatibility for WebResource Module

### PR DESCRIPTION
**Pull request by M-Barz**

### CODE REVIEW

File: Modules/WebResource/classes/class.ilLinkResourceHandlerGUI.php [hunk 1]
```diff
@@ -68,18 +69,17 @@ public function executeCommand()
         }
 
         // add entry to navigation history
-        if ($ilAccess->checkAccess("read", "", $_GET["ref_id"])) {
+        if ($ilAccess->checkAccess("read", "", $getParams["ref_id"])) {
```
Comment (smeyer-ilias) [inline Modules/WebResource/classes/class.ilLinkResourceHandlerGUI.php]: Theoretically the index "ref_id" is not defined

File: Modules/WebResource/classes/class.ilLinkResourceHandlerGUI.php [hunk 2]
```diff
@@ -68,18 +69,17 @@ public function executeCommand()
...
             $ilNavigationHistory->addItem(
-                $_GET["ref_id"],
-                "ilias.php?baseClass=ilLinkResourceHandlerGUI&cmd=infoScreen&ref_id=" . $_GET["ref_id"],
+                $getParams["ref_id"],
```
Comment (smeyer-ilias) [inline Modules/WebResource/classes/class.ilLinkResourceHandlerGUI.php]: Same here

File: Modules/WebResource/classes/class.ilObjLinkResourceGUI.php [hunk 3]
```diff
@@ -30,26 +27,45 @@ class ilObjLinkResourceGUI extends ilObject2GUI implements ilLinkCheckerGUIRowHa
...
-        $ilTabs = $DIC['ilTabs'];
-        $ilErr = $DIC['ilErr'];
-        $ilAccess = $DIC['ilAccess'];
-        
-        
-        //if($this->ctrl->getTargetScript() == 'link_resources.php')
-        if ($_GET["baseClass"] == 'ilLinkResourceHandlerGUI') {
-            $_GET['view_mode'] = isset($_GET['switch_mode']) ? $_GET['switch_mode'] : $_GET['view_mode'];
-            $ilCtrl->saveParameter($this, 'view_mode');
+        if ($this->getParams["baseClass"] == 'ilLinkResourceHandlerGUI') {
```
Comment (smeyer-ilias) [inline Modules/WebResource/classes/class.ilObjLinkResourceGUI.php]: Please check if "baseClass" exists.

File: Modules/WebResource/classes/class.ilObjLinkResourceGUI.php [hunk 4]
```diff
@@ -347,44 +338,36 @@ protected function initFormSettings()
...
     {
-        global $DIC;
-
-        $ilCtrl = $DIC['ilCtrl'];
-        
         $this->checkPermission('write');
         $this->activateTabs('content', 'id_content_view');
         
-        if (!(int) $_GET['link_id']) {
+        if (!(int) $this->getParams['link_id']) {
```
Comment (smeyer-ilias) [inline Modules/WebResource/classes/class.ilObjLinkResourceGUI.php]: Please check whether "link_id" is set.

File: Modules/WebResource/classes/class.ilObjLinkResourceGUI.php [hunk 5]
```diff
@@ -347,44 +338,36 @@ protected function initFormSettings()
...
             ilUtil::sendFailure($this->lng->txt('select_one'), true);
-            $ilCtrl->redirect($this, 'view');
+            $this->ctrl->redirect($this, 'view');
         }
         
         $this->initFormLink(self::LINK_MOD_EDIT);
-        $this->setValuesFromLink((int) $_GET['link_id']);
+        $this->setValuesFromLink((int) $this->getParams['link_id'] ?? 0);
```
Comment (smeyer-ilias) [inline Modules/WebResource/classes/class.ilObjLinkResourceGUI.php]: Precedence (parenthesis)

File: Modules/WebResource/classes/class.ilObjLinkResourceGUI.php [hunk 6]
```diff
@@ -519,82 +494,68 @@ public function saveAddLink()
...
     protected function deleteParameter()
     {
-        global $DIC;
-
-        $ilCtrl = $DIC['ilCtrl'];
-
         $this->checkPermission('write');
         
-        $this->ctrl->setParameter($this, 'link_id', (int) $_GET['link_id']);
+        $this->ctrl->setParameter($this, 'link_id', (int) $this->getParams['link_id'] ?? 0);
```
Comment (smeyer-ilias) [inline Modules/WebResource/classes/class.ilObjLinkResourceGUI.php]: Precedence (parenthesis)

File: Modules/WebResource/classes/class.ilObjLinkResourceGUI.php [hunk 7]
```diff
@@ -519,82 +494,68 @@ public function saveAddLink()
...
+        if (!isset($this->getParams['param_id'])) {
             ilUtil::sendFailure($this->lng->txt('select_one'), true);
-            $ilCtrl->redirect($this, 'view');
+            $this->ctrl->redirect($this, 'view');
         }
 
-        include_once './Modules/WebResource/classes/class.ilParameterAppender.php';
         $param = new ilParameterAppender($this->object->getId());
-        $param->delete((int) $_GET['param_id']);
+        $param->delete((int) $this->getParams['param_id'] ?? 0);
```
Comment (smeyer-ilias) [inline Modules/WebResource/classes/class.ilObjLinkResourceGUI.php]: Precedence (parenthesis)

File: Modules/WebResource/classes/class.ilObjLinkResourceGUI.php [hunk 8]
```diff
@@ -519,82 +494,68 @@ public function saveAddLink()
...
```
Comment (smeyer-ilias) [inline Modules/WebResource/classes/class.ilObjLinkResourceGUI.php]: Precedence (parenthesis)

File: Modules/WebResource/classes/class.ilObjLinkResourceGUI.php [hunk 9]
```diff
@@ -519,82 +494,68 @@ public function saveAddLink()
...
     {
-        global $DIC;
-
-        $ilCtrl = $DIC['ilCtrl'];
         
         $this->checkPermission('write');
         $this->activateTabs('content', '');
         
-        if (!is_array($_POST['ids'])) {
+        if (!is_array($this->postParams['ids'])) {
```
Comment (smeyer-ilias) [inline Modules/WebResource/classes/class.ilObjLinkResourceGUI.php]: Check  for "ids"

File: Modules/WebResource/classes/class.ilObjLinkResourceGUI.php [hunk 10]
```diff
@@ -519,82 +494,68 @@ public function saveAddLink()
...
             ilUtil::sendFailure($this->lng->txt('select_one'), true);
-            $ilCtrl->redirect($this, 'view');
+            $this->ctrl->redirect($this, 'view');
         }
         
         // Validate
         $invalid = array();
-        foreach ($_POST['ids'] as $link_id) {
-            $data = $_POST['links'][$link_id];
+        foreach ($this->postParams['ids'] as $link_id) {
```
Comment (smeyer-ilias) [inline Modules/WebResource/classes/class.ilObjLinkResourceGUI.php]: Check index defined for the following lines.

File: Modules/WebResource/classes/class.ilObjLinkResourceGUI.php [hunk 11]
```diff
@@ -619,24 +580,21 @@ protected function updateLinks()
         if (count($invalid)) {
             ilUtil::sendFailure($this->lng->txt('err_check_input'));
             $this->tpl->addBlockFile('ADM_CONTENT', 'adm_content', 'tpl.webr_manage.html', 'Modules/WebResource');
-            
-            include_once './Modules/WebResource/classes/class.ilWebResourceEditableLinkTableGUI.php';
+
             $table = new ilWebResourceEditableLinkTableGUI($this, 'view');
             $table->setInvalidLinks($invalid);
-            $table->parseSelectedLinks($_POST['ids']);
+            $table->parseSelectedLinks( (array) $this->postParams['ids'] ?? array());
```
Comment (smeyer-ilias) [inline Modules/WebResource/classes/class.ilObjLinkResourceGUI.php]: Precedence (parenthesis)

File: Modules/WebResource/classes/class.ilObjLinkResourceGUI.php [hunk 12]
```diff
@@ -619,24 +580,21 @@ protected function updateLinks()
...
-        include_once './Modules/WebResource/classes/class.ilLinkResourceItems.php';
+
         $links = new ilLinkResourceItems($this->object->getId());
         
         // Save Settings
-        include_once './Services/Form/classes/class.ilFormPropertyGUI.php';
-        include_once './Services/Form/classes/class.ilLinkInputGUI.php';
-        foreach ($_POST['ids'] as $link_id) {
-            $data = $_POST['links'][$link_id];
+        $ids = (array) $this->postParams['ids'] ?? array();
```
Comment (smeyer-ilias) [inline Modules/WebResource/classes/class.ilObjLinkResourceGUI.php]: Precedence (parenthesis)
And following lines

File: Modules/WebResource/classes/class.ilObjLinkResourceGUI.php [hunk 13]
```diff
@@ -920,7 +860,7 @@ protected function initFormLink($a_mode)
                 $dyn->setInfo($this->lng->txt('links_dynamic_info'));
 
 
-                if (count($links = ilParameterAppender::_getParams((int)$_GET['link_id']))) {
+                if (count($links = ilParameterAppender::_getParams( (int) $this->getParams['link_id'] ?? 0))) {
```
Comment (smeyer-ilias) [inline Modules/WebResource/classes/class.ilObjLinkResourceGUI.php:872]: Precedence (parenthesis)

File: Modules/WebResource/classes/class.ilObjLinkResourceGUI.php [hunk 14]
```diff
@@ -964,21 +896,15 @@ protected function initFormLink($a_mode)
...
-     * @return
      */
     protected function switchViewMode()
     {
-        global $DIC;
-
-        $ilCtrl = $DIC['ilCtrl'];
-        
-        $_REQUEST['view_mode'] = $_GET['view_mode'] = (int) $_GET['switch_mode'];
+        $_REQUEST['view_mode'] = $this->getParams['view_mode'] = (int) $this->getParams['switch_mode'] ?? self::VIEW_MODE_VIEW;
```
Comment (smeyer-ilias) [inline Modules/WebResource/classes/class.ilObjLinkResourceGUI.php]: Precedence (parenthesis)

File: Modules/WebResource/classes/class.ilObjLinkResourceGUI.php [hunk 15]
```diff
@@ -989,31 +915,25 @@ protected function editLinks()
...
-        $ilErr = $DIC['ilErr'];
-        $ilTabs = $DIC['ilTabs'];
         
-        $ilTabs->activateTab("id_content");
+        $this->tabs_gui->activateTab("id_content");
         
         $this->checkPermission('read');
         
-        if (strtolower($_GET["baseClass"]) == "iladministrationgui") {
+        if (strtolower($this->getParams["baseClass"]) == "iladministrationgui") {
```
Comment (smeyer-ilias) [inline Modules/WebResource/classes/class.ilObjLinkResourceGUI.php:937]: Check "baseClass" exists.

File: Modules/WebResource/classes/class.ilObjLinkResourceGUI.php [hunk 16]
```diff
@@ -1093,15 +1006,13 @@ protected function sort()
...
      */
     protected function saveSorting()
     {
         $this->checkPermission('write');
-        
-        include_once './Services/Container/classes/class.ilContainerSorting.php';
+
         $sort = ilContainerSorting::_getInstance($this->object->getId());
-        $sort->savePost((array) $_POST['position']);
+        $sort->savePost((array) $this->postParams['position'] ?? array());
```
Comment (smeyer-ilias) [inline Modules/WebResource/classes/class.ilObjLinkResourceGUI.php:1026]: Parenthesis

File: Modules/WebResource/classes/class.ilObjLinkResourceGUI.php [hunk 17]
```diff
@@ -1158,19 +1067,18 @@ protected function confirmDeleteLink()
         
         $link_ids = array();
 
-        if (is_array($_POST['link_ids'])) {
-            $link_ids = $_POST['link_ids'];
-        } elseif (isset($_GET['link_id'])) {
-            $link_ids = array($_GET['link_id']);
+        if (is_array($this->postParams['link_ids'])) {
```
Comment (smeyer-ilias) [inline Modules/WebResource/classes/class.ilObjLinkResourceGUI.php:1081]: Parenthesis

File: Modules/WebResource/classes/class.ilObjLinkResourceGUI.php [hunk 18]
```diff
@@ -1184,55 +1092,47 @@ protected function confirmDeleteLink()
...
         
         $this->checkPermission('write');
-        
-        include_once './Modules/WebResource/classes/class.ilLinkResourceItems.php';
+
         $links = new ilLinkResourceItems($this->object->getId());
-        
-        foreach ($_POST['link_ids'] as $link_id) {
+
+        $link_ids = (array) $this->postParams['link_ids'] ?? array();
```
Comment (smeyer-ilias) [inline Modules/WebResource/classes/class.ilObjLinkResourceGUI.php:1115]: Parenthesis

File: Modules/WebResource/classes/class.ilObjLinkResourceGUI.php [hunk 19]
```diff
@@ -1184,55 +1092,47 @@ protected function confirmDeleteLink()
...
-        $ilCtrl = $DIC['ilCtrl'];
         
         $this->checkPermission('write');
-        
-        include_once './Modules/WebResource/classes/class.ilLinkResourceItems.php';
+
         $links = new ilLinkResourceItems($this->object->getId());
 
-        if (!$_GET['link_id']) {
+        if (!$this->getParams['link_id']) {
```
Comment (smeyer-ilias) [inline Modules/WebResource/classes/class.ilObjLinkResourceGUI.php:1132]: Check "link_id" exists

File: Modules/WebResource/classes/class.ilObjLinkResourceGUI.php [hunk 20]
```diff
@@ -1184,55 +1092,47 @@ protected function confirmDeleteLink()
...
             ilUtil::sendFailure($this->lng->txt('select_one'), true);
-            $ilCtrl->redirect($this, 'view');
+            $this->ctrl->redirect($this, 'view');
         }
         
-        $links->setLinkId((int) $_GET['link_id']);
+        $links->setLinkId((int) $this->getParams['link_id'] ?? 0);
```
Comment (smeyer-ilias) [inline Modules/WebResource/classes/class.ilObjLinkResourceGUI.php]: Parenthesis

File: Modules/WebResource/classes/class.ilObjLinkResourceGUI.php [hunk 21]
```diff
@@ -1367,23 +1245,19 @@ protected function linkChecker()
...
-
-        include_once './Services/LinkChecker/classes/class.ilLinkCheckNotify.php';
-
         $link_check_notify = new ilLinkCheckNotify($ilDB);
-        $link_check_notify->setUserId($ilUser->getId());
+        $link_check_notify->setUserId($this->user->getId());
         $link_check_notify->setObjId($this->object->getId());
 
-        if ($_POST['link_check_message']) {
+        if ($this->postParams['link_check_message']) {
```
Comment (smeyer-ilias) [inline Modules/WebResource/classes/class.ilObjLinkResourceGUI.php]: Check link_check_message" exists.

File: Modules/WebResource/classes/class.ilObjLinkResourceListGUI.php [hunk 22]
```diff
@@ -132,9 +113,24 @@ public function getProperties()
     */
     public function getCommandLink($a_cmd)
     {
+        global $DIC;
+        $request = $DIC->http()->request();
+
+        if(isset($request->getQueryParams()['wsp_id'])) {
```
Comment (smeyer-ilias) [inline Modules/WebResource/classes/class.ilObjLinkResourceListGUI.php:120]: Please initialize "wsp_id"

File: Modules/WebResource/classes/class.ilObjLinkResourceListGUI.php [hunk 23]
```diff
@@ -132,9 +113,24 @@ public function getProperties()
...
+            $cmd_class = $request->getQueryParams()['cmdClass'];
+        } else if(isset($request->getParsedBody()['cmdClass'])) {
+            $cmd_class = $request->getParsedBody()['cmdClass'];
+        }
+
         if (
-            (isset($_REQUEST["wsp_id"]) && $_REQUEST["wsp_id"]) ||
-            (isset($_REQUEST["cmdClass"]) && $_REQUEST["cmdClass"] === "ilpersonalworkspacegui")
+            (isset($wsp_id) && $wsp_id) ||
+            (isset($cmd_class) && $cmd_class === "ilpersonalworkspacegui")
```
Comment (smeyer-ilias) [inline Modules/WebResource/classes/class.ilObjLinkResourceListGUI.php]: Please initialize $cmd_class

File: Modules/WebResource/classes/class.ilParameterAppender.php [hunk 24]
```diff
@@ -21,15 +21,15 @@
     +-----------------------------------------------------------------------------+
 */
 
-define("LINKS_USER_ID", 1);
-define("LINKS_SESSION_ID", 2);
-define("LINKS_LOGIN", 3);
-define('LINKS_MATRICULATION', 4);
+const LINKS_USER_ID = 1;
```
Comment (smeyer-ilias) [inline Modules/WebResource/classes/class.ilParameterAppender.php]: Please move to ilParameterAppender as public const

**Review by smeyer-ilias — CHANGES_REQUESTED** [review]

**Comment by mjansenDatabay** [comment]
Hi @M-Barz 

we stumbled upon this PR in our TB meeting. Any updates on this?

Best regards!

**Comment by Amstutz** [comment]
Hi @M-Barz

we stumbled upon this PR in our TB meeting. Any updates on this? Can that PR be closed be closed?

Best regards!

**Comment by smeyer-ilias** [comment]
Already integrated in trunk.

**Pull request closed without merging**